README.md

```diff
@@ -1,10 +1,204 @@
+
+```
+# Clone the repository
+git clone https://github.com/AndrewAnnex/planetcantile.git
+cd planetcantile
+
+# Install with web application (Includes the core library plus dependencies)
+pip install -e ".[app]"
+
+# Or install for development (Includes all dependencies plus development tools)
+pip install -e ".[dev]"
+
+# Then launch the application (with auto-reload for development)
+uvicorn planetcantile.app:app --host 0.0.0.0 --port 8000 --reload
+```
+
+#### Docker Container (Experimental)
+
+Warning: The Docker container is currently experimental and may crash unexpectedly. It can be useful for quick exploration but is not recommended for production use: [planetcantile_docker](https://github.com/AndrewAnnex/planetcantile_docker)
+
+## Usage
+
+### Option 1: Using the Web API
+
+After starting the web application, the API interface will be accessible at [http://localhost:8000/docs](http://localhost:8000/docs)
+
+More information about endpoints is available in the [TiTiler documentation](https://developmentseed.org/titiler/).
+
+Example tile request:
+
+```
+http://localhost:8000/cog/tiles/MarsGeographicSphere/{z}/{x}/{y}?url=https://path/to/mars.tif
+```
+
+For local files, replace the URL with the local file path: ``url=file:///home/user/path/to/mars.tif``.
+
+### Option 2: As a Python Library
+
+
+```python
+# Import the planetary TMS collection
+from planetcantile import planetary_tms
+
+# List available TMS
+print(planetary_tms.list())
+
+# Get a specific TMS
+mars_tms = planetary_tms.get("MarsGeographicSphere")
+
+# Register a custom TMS (example)
+# planetary_tms.register(custom_tms)
+
+# Use with morecantile
+from morecantile import Tile
+tile = Tile(x=0, y=0, z=0)
+bounds = mars_tms.bounds(tile)
+print(f"Bounds of Mars tile: {bounds}")
+```
+
+### Environment Variables
+
+<strong>TILEMATRIXSET_DIRECTORY</strong>
+This environment variable allows you to specify a directory containing custom TileMatrixSet JSON definitions. Planetcantile will merge these with its own TMS definitions.
+
+You can set this programmatically before importing planetcantile:
+
+```python
+import os
+
+# Set the environment variable
+os.environ["TILEMATRIXSET_DIRECTORY"] = "/path/to/your/custom/tms/definitions"
+
+# Import planetcantile - it will include your custom TMS
+from planetcantile import planetary_tms
+```
+
+### Extending with Custom Definitions
+
+Planetcantile supports two approaches for adding custom TMS definitions:
+
+1. <strong>Using the generate.py script:</strong> This gives you fine-grained control over how the TMS definitions are generated, particularly when working with complex planetary coordinate reference systems.
+
+2. <strong>Adding JSON files directly:</strong> As a simpler alternative, you can create TMS JSON definition files and add them to your custom directory specified by the ``TILEMATRIXSET_DIRECTORY`` environment variable.
+
+## Supported Planetary Bodies
+
+Planetcantile includes TMS definitions for:
+
```

Comment:
probably good to mention that planetcantile defines TMSs based on official IAU CRSs registered with the OGC listed at http://voparis-vespa-crs.obspm.fr:8080/web/